src/main/java/com/exasol/adapter/document/documentfetcher/files/JsonDocumentFetcher.java

```diff
@@ -17,26 +18,39 @@
  * {@link DocumentFetcher} for JSON files.
  */
 public class JsonDocumentFetcher implements DocumentFetcher<JsonNodeVisitor> {
-    private final String fileName;
+    private static final long serialVersionUID = 1252243324748529337L;
+    private final String filePattern;
+    SegmentDescription segmentDescription;
 
     public static final String FILE_EXTENSION = ".json";
 
     /**
      * Create an instance of {@link JsonDocumentFetcher}.
      * 
-     * @param fileName file pattern
+     * @param filePattern        files to load
+     * @param segmentDescription segmentation for parallel execution
      */
-    public JsonDocumentFetcher(final String fileName) {
-        this.fileName = fileName;
+    public JsonDocumentFetcher(final String filePattern, final SegmentDescription segmentDescription) {
+        this.filePattern = filePattern;
+        this.segmentDescription = segmentDescription;
     }
 
     @Override
     public Stream<DocumentNode<JsonNodeVisitor>> run(final ExaConnectionInformation connectionInformation) {
-        final InputStream jsonStream = FileLoaderFactory.getInstance().getLoader(this.fileName, connectionInformation)
-                .loadFile();
+        final Stream<InputStream> jsonStream = FileLoaderFactory.getInstance()
+                .getLoader(this.filePattern, this.segmentDescription, connectionInformation).loadFiles();
+        return jsonStream.map(this::readJson);
+    }
+
+    private DocumentNode<JsonNodeVisitor> readJson(final InputStream jsonStream) {
         try (final JsonReader jsonReader = Json.createReader(jsonStream)) {
             final JsonValue jsonValue = jsonReader.readValue();
-            return Stream.of(JsonNodeFactory.getInstance().getJsonNode(jsonValue));
+            try {
+                jsonStream.close();
```

Comment:
The input stream already consumed at this point, right?

Reply:
yes